site/app/controllers/admin/StudentActivityDashboardController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace app\controllers\admin;
+
+use app\controllers\AbstractController;
+use app\libraries\routers\AccessControl;
+use Symfony\Component\Routing\Annotation\Route;
+use app\libraries\response\WebResponse;
+use app\libraries\FileUtils;
+
+/**
+ * Class StudentActivityDashboardController
+ * @package app\controllers\admin
+ */
+
+class StudentActivityDashboardController extends AbstractController {
+  /**
+   * @Route("/courses/{_semester}/{_course}/activity", methods={"GET"})
+   * @AccessControl(role="INSTRUCTOR")
+   */
+    public function getStudents() {
+        $data_dump = $this->core->getQueries()->getAttendanceInfo();
+        return new WebResponse([
+            'admin',
+            'StudentActivityDashboard'
+        ], 'createTable', $data_dump);
+    }
+
+   /**
+    * @Route("/courses/{_semester}/{_course}/activity/download", methods={"GET"})
+    * @AccessControl(role="INSTRUCTOR")
+    */
+    public function downloadData() {
```

Comment:
This function should include column names or else you can't tell what you are looking at in CSV.